### PR TITLE
feat!: replace old demo course with the new one

### PR DESCRIPTION
**Pull request by kdmccormick**
This subsumes the [openedx-demo-course-new](https://github.com/openedx/openedx-demo-course-new) into this repo, and removes the entirety of the old demo course. We will archive openedx-demo-course-new after merging this PR.

Notes for reviewers:

* I made this PR by rm'ing the current repository and replacing it with the contents of https://github.com/openedx/openedx-demo-course-new, plus just a couple tweaks:
  * I changed the GHA CI branches from `main` back to `master`.
  * I put openedx.yaml back so this this repo continues to be released.
  * I fixed the repo name in catalog-info.yaml
* The insane -400k diff is probably because the jsmol library had been entirely copied into this repo previously.
* Rather than trying to read the diff, I would just [browse the branch a bit](https://github.com/openedx/openedx-demo-course/tree/kdmccormick/new-demo-course) as a sanity check.
 
### Operator Impact

On its own, this PR will break the `tutor local do importdemocourse` command in Tutor Nightly. Merging [this update to Tutor Nightly](https://github.com/overhangio/tutor/pull/976) will fix the command again. With that PR merged:

* For Quince (and earlier) users, the default  remains the old course. Specifically:
  * Get the old course with `tutor local do importdemocourse`
  * Get the new course with: `tutor local do importdemocourse demo-course/course --version master && tutor local do importdemolibraries --version master`
* For master and Redwood+ users, the default becomes the new course. Specifically:
  * Get the old course with: `tutor local do importdemocourse --version open-release/quince.1`
  * Get the new course and library with: `tutor local do importdemocourse && tutor local do importdemolibraries`

### CODE REVIEW

**Comment by kdmccormick** [comment]
@sarina One last PR for you! With this one merged and the linked Tutor PR merged, we'll be fully prepped to switch over to the new demo course for Redwood.

File: .github/workflows/ci.yml
Comment (sarina) [inline .github/workflows/ci.yml:7]: should this be "main"?

Comment (sarina) [reply .github/workflows/ci.yml:7]: nevermind I misunderstood this PR, read more carefully now. GTG

**Comment by sarina** [comment]
This looks good, before I give thumbs-up I want to do a quick test of the tar.gz into my personal instance, which I'll do now

**Comment by sarina** [comment]
This should be updated with the changes made in https://github.com/openedx/openedx-demo-course-new/pull/4 - once that's done we can go ahead and merge.

**Comment by kdmccormick** [comment]
@sarina I pushed two commits, ready for your review:

1. Copy in @jswope00 's recent changes: https://github.com/openedx/openedx-demo-course/pull/47/commits/ab4a57a11c46fbfbd3479e059e64d698bd1d5a88
2. Remove some macOS hidden-file-cruft from the OLX tars, and update the Makefile so that cruft doesn't get added to the tars next time: https://github.com/openedx/openedx-demo-course/pull/47/commits/59e63209f2b66c7b405ccee3f7590730d384a810

I just need to address [Régis's comments over on the Tutor PR](https://github.com/overhangio/tutor/pull/976), and then I think we're all set!

**Review by sarina — APPROVED** [review]
Thanks Kyle, merge away when ready!

**Comment by sarina** [comment]
Hey @kdmccormick - we merged in one more change to https://github.com/openedx/openedx-demo-course-new - it'd be good to get this updated & merged, so we can close down the openedx-demo-course-new repo and churn on this PR. Let me know the best way to proceed - should we merge this and port the changes from the last PR in a separate PR?

**Comment by kdmccormick** [comment]
@sarina I was holding off on merging this because it'll break `importdemocourse` in Tutor Nightly unless the Tutor PR is merged first. The drift between this and openedx-demo-course-new isn't good, though, so I'm going to merge this now and the Tutor fix will be a fast follow.

**Pull request merged**
